presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
```

Comment:
according to docs: http://dev.mysql.com/doc/refman/5.7/en/char.html, plain varchar can take up to  65,535 characters